packages/gafl-webapp-service/src/handlers/oidc-handler.js

```diff
@@ -0,0 +1,108 @@
+import { salesApi } from '@defra-fish/connectors-lib'
+import { OIDC_SIGNIN } from '../uri.js'
+import Boom from '@hapi/boom'
+import OpenIdClient from 'openid-client'
+import moment from 'moment'
+import debug from 'debug'
+
+const { generators, Issuer } = OpenIdClient
+const db = debug('webapp:oidc')
+
+let client = null
+let cache = null
+let redirectUri = null
+/**
+ * Initialise OpenID connect authentication for the hapi server
+ *
+ * @param server the hapi server instance to configure
+ * @returns {Promise<void>}
+ */
+export const initialise = async server => {
+  const MsIssuer = await Issuer.discover(process.env.OIDC_ENDPOINT)
+  db('Discovered issuer %s %O', MsIssuer.issuer, MsIssuer.metadata)
+
+  redirectUri = new URL(OIDC_SIGNIN.uri, process.env.OIDC_REDIRECT_HOST).toString()
+  client = new MsIssuer.Client({
+    client_id: process.env.OIDC_CLIENT_ID,
+    client_secret: process.env.OIDC_CLIENT_SECRET,
+    redirect_uris: [redirectUri],
+    response_types: ['code', 'id_token']
+  })
+  cache = server.cache({ segment: 'oidc', expiresIn: 12 * 60 * 60 * 1000 })
```

Comment:
Should extract this to an environment variable? Security will try and get us to reduce it. -Should be synchronised with the ttl on the cookie in the same way as the session cookie. I think I would have integrated this with the existing session management.

Reply:
This is a server-side cache which is used to store data while the user has been redirected to the OpenID connect endpoint, it is not the lifetime of the authenticated OIDC session (that is set below by examining the exp claim of the id_token returned when successfully authenticated).

This cache merely stores a randomly generated state and nonce value along with the page url requested before the authentication redirect occurred.  I know 12 hours seems a little arbitrary but this seemed like a reasonable value as the data stored here is very small and we don't want telesales users encountering an error if they sit on authentication page for several hours and then later enter the username/password.